src/Cursor.js

```diff
@@ -1,41 +1,34 @@
-var React = require('react/addons');
-var util = require('./util');
+import React from 'react/addons';
+import util from './util';
 
 'use strict';
 
-function Cursor(cmp, path, value) {
+function Cursor(state, swapper, path, value) {
   // value to put in the DOM, use from render() and the component lifecycle methods
   this.value = value;
 
-  this.onChange = function (nextValue) {
-    if (Cursor.debug === true) {
-      console.warn("'onChange' is deprecated use 'set' instead!");
-    }
-    this.set(nextValue);
-  };
-
   ['push', 'unshift', 'splice', 'set', 'merge', 'apply'].forEach(function (command) {
-    this[command] = update.bind(this, cmp, path, '$' + command);
+    this[command] = update.bind(this, swapper, path, '$' + command);
   }.bind(this));
 
   this.refine = function (/* one or more paths through the tree */) {
-    // When refining inside a lifecycle method, same cmp and same path isn't enough.
+    // When refining inside a lifecycle method, same cmp(swapper) and same path isn't enough.
     // this.props and nextProps have different subtree values, and refining memoizer must account for that
 
     var nextPath = [].concat(path, util.flatten(arguments));
     var nextValue = util.getRefAtPath(this.value, Array.prototype.slice.call(arguments, 0));
-    return build(cmp, nextPath, nextValue); // memoized
+    return build(state, swapper, nextPath, nextValue); // memoized
   };
 }
 
-function update(cmp, path, operation, nextUpdate) {
+function update(swapper, path, operation, nextUpdate) {
   // Backwards compatibility with non-function values of nextUpdate
   if (typeof nextUpdate !== "function") {
     var prevValue = nextUpdate;
     nextUpdate = function ( ) { return prevValue; };
   }
 
-  cmp.setState(function (state) {
+  swapper(function (state) {
```

Comment:
Is your `swapper` function finally used as `setState` here? If so, why not directly name as `setState`?